components/nginx/scripts/preconfigure.py

```diff
@@ -118,14 +151,34 @@ def preconfigure_nginx():
 
 def create_certs():
     utils.mkdir(utils.SSL_CERTS_TARGET_DIR)
-    utils.generate_ca_cert()
+    ca_cert_deployed, ca_key_deployed = _deploy_cert_and_key(
+        'ca_certificate', 'ca_key',
+        utils.INTERNAL_CA_CERT_PATH, utils.INTERNAL_CA_KEY_PATH)
+    has_ca_key = ca_key_deployed
+    if not ca_cert_deployed:
+        if ca_key_deployed:
+            ctx.abort_operation('Internal CA key provided, but the internal '
+                                'CA cert was not')
+        utils.generate_ca_cert()
+        has_ca_key = True
+
     networks = \
         ctx.target.node.properties['cloudify']['cloudify_agent']['networks']
     internal_rest_host = \
         ctx.target.instance.runtime_properties['internal_rest_host']
     utils.store_cert_metadata(internal_rest_host, networks)
-    cert_ips = [internal_rest_host] + list(networks.values())
-    utils.generate_internal_ssl_cert(ips=cert_ips, name=internal_rest_host)
+
+    internal_cert_deployed, internal_key_deployed = _deploy_cert_and_key(
+        'internal_certificate', 'internal_key',
+        utils.INTERNAL_CERT_PATH, utils.INTERNAL_KEY_PATH)
+
+    if not internal_cert_deployed and not internal_key_deployed:
```

Comment:
Ditto re comments.

Reply:
added a few comments (didn't affect this very line so github won't hide this message)